internal/cmd/docker/push.go

```diff
@@ -0,0 +1,129 @@
+package docker
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/client"
+	cmds "github.com/saucelabs/saucectl/internal/cmd"
+	"github.com/saucelabs/saucectl/internal/segment"
+	"github.com/saucelabs/saucectl/internal/usage"
+	"github.com/spf13/cobra"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+func PushCommand() *cobra.Command {
+	var registryPushTimeout time.Duration
+	var quiet bool
+
+	cmd := &cobra.Command{
+		Use:          "push <IMAGE_NAME>",
+		Short:        "Push a Docker image to the Sauce Labs Container Registry.",
+		SilenceUsage: true,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 || args[0] == "" {
+				return errors.New("no docker image specified")
+			}
+
+			return nil
+		},
+		PreRun: func(cmd *cobra.Command, args []string) {
+			tracker := segment.DefaultTracker
+
+			go func() {
+				tracker.Collect(
+					cases.Title(language.English).String(cmds.FullName(cmd)),
+					usage.Properties{}.SetFlags(cmd.Flags()),
+				)
+				_ = tracker.Close()
+			}()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			image := args[0]
+			repo, err := extractRepo(image)
+			if err != nil {
+				return err
+			}
+			auth, err := imageRunnerService.RegistryLogin(context.Background(), repo)
+			if err != nil {
+				return fmt.Errorf("failed to fetch auth token: %v", err)
+			}
+			return pushDockerImage(image, auth.Username, auth.Password, registryPushTimeout, quiet)
+		},
+	}
+
+	flags := cmd.PersistentFlags()
+	flags.DurationVar(&registryPushTimeout, "registry-push-timeout", 5*time.Minute, "Configure the timeout duration for docker push . Default: 5 minute.")
```

Comment:
```suggestion
	flags.DurationVar(&registryPushTimeout, "registry-push-timeout", 5*time.Minute, "Configure the timeout duration for docker push. Default: 5 minutes.")
```

Also, does `registry-push-timeout` need to be this long? It's set within the context of `saucectl docker push {bla} --registry-push-timeout`. Can there be a different push than to a registry within this context? If not, then you can shorten it to `--push-timeout`. In fact, you can likely shorten it to `--timeout`, because in the context of `saucectl docker push`, _push_ is the main process that takes up the longest chunk of time. If there's a need for additional configurable timeouts, those would be prefixed (e.g. `--auth-timeout`).

You can also just use it as a global timeout for the entire command with its `Context` that is pushed down to all the clients used here. That way, the entire command has 5 minutes to complete, regardless of what it's doing underneath.